### PR TITLE
WIP initial groups index page

### PR DESCRIPTION
**Pull request by zehicle**
Available under the Edge UI flag only, this renders the groups index

Connect to #732 

Signed-off-by: robhirschfeld <rob@zehicle.com>

### CODE REVIEW

**Comment by galthaus** [comment]
This Pull Request needs a rebase to be pulled.

**Comment by zehicle** [comment]
Reminder: this function is hidden behind the EDGE UI flag so no one sees it by default.
You can also go tot the /groups page.

You can add groups to the system.  If you add a new type of group it will automatically put all the nodes into that group.  When you add a second group then you can start moving nodes into that other group.

The workflow is a little off but it's enough to get feedback.

I also made sure that ansible now sees the groups.

File: rails/app/models/node.rb [hunk 1]
```diff
@@ -197,7 +198,7 @@ def merge_quirks(new_quirks)
   end
 
   def active_node_roles
-    NodeRole.on_node(self).in_state(NodeRole::ACTIVE).committed.order("cohort ASC")
+    NodeRole.on_node(setlf).in_state(NodeRole::ACTIVE).committed.order("cohort ASC")
```
Comment (galthaus) [inline rails/app/models/node.rb]: This should probably be self and not setlf

Comment (zehicle) [inline rails/app/models/node.rb]: fixed

On 08/12/2015 09:39 AM, Greg Malthus wrote:
>
> In rails/app/models/node.rb
> <https://github.com/opencrowbar/core/pull/773#discussion_r36859829>:
>
> > @@ -197,7 +198,7 @@ def merge_quirks(new_quirks)
> >    end
> >  
> >    def active_node_roles
> > -    NodeRole.on_node(self).in_state(NodeRole::ACTIVE).committed.order("cohort ASC")
> > +    NodeRole.on_node(setlf).in_state(NodeRole::ACTIVE).committed.order("cohort ASC")
>
> This should probably be self and not setlf
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencrowbar/core/pull/773/files#r36859829>.
>

-- 
  

Rob
____________________________
Rob Hirschfeld, 512-773-7522
RackN CEO, Founder

I am in CENTRAL (-6) time
http://robhirschfeld.com
twitter: @zehicle, github: cloudedge & ravolt

**Comment by zehicle** [comment]
I figured out some better UI workflow.... patch coming

**Comment by ravolt** [comment]
this should be ready to go for user feedback considering it is under the Edge UI flag still.

File: rails/app/models/node.rb (hunk 1 above)
Comment (zehicle) [inline rails/app/models/node.rb]: Hi,

I'd like to add you to my professional network on LinkedIn.

- Rob

Confirm that you know Rob: https://www.linkedin.com/e/v2?e=0-im6czafy-bp&t=ssuw&tracking=eml-guest-invite-cta&ek=invite_guest&sharedKey=eWcsnCis&invitationId=6118771214266753024








You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=0-im6czafy-bp&t=lun&midToken=AQFmrFVhGEd36Q&ek=invite_guest&loid=AQFUu2JU-lCrtwAAAVOo-Gxh2wTrDGUQWr8JlnnXRsrpDi0D16LSBP5LQypTeltCmh6X-VU3nH7Tvdj2U_GmkxvYBrjY7NlIF7jAnc-WB985IyjyXog6Juvnfru3UaBE5D7jZKxQMwQlpvizSfn1vBwvmXmz4kegiokLEFI8lef3KonjuGFGesyTD8yzSaQVFBUE&eid=0-im6czafy-bp


This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=0-im6czafy-bp&a=customerServiceUrl&ek=invite_guest

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

Comment (zehicle) [inline rails/app/models/node.rb]: Rob Hirschfeld would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-1sdvr2-imfatvhn-s5&a=preRegInvite&ek=first_guest_reminder_01&li=14&m=hero&ts=accept_text&sharedKey=eWcsnCis&invitationID=6118771214266753024

View Rob Hirschfeld&#39;s profile: https://www.linkedin.com/e/v2?e=-1sdvr2-imfatvhn-s5&a=preRegInvite&ek=first_guest_reminder_01&li=3&m=hero&ts=profile_text&sharedKey=eWcsnCis&invitationID=6118771214266753024





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-1sdvr2-imfatvhn-s5&t=lun&midToken=AQFmrFVhGEd36Q&ek=first_guest_reminder_01&li=16&m=unsub&ts=HTML&eid=-1sdvr2-imfatvhn-s5&loid=AQH3YORYM2LSWgAAAVPJ-K76nqLa1JJ3bpSmuo7qsiTShq2Ru12T8PpjQhqPsCq1FZxkuIc0hqkn2U9aPxC66OtMJv9bnXl9VEcL930w8mh1cy-9rOweKXN9QEgwIwtV4sQj32v73wYVFB7BKpnFGe2yxL-v-FSFTol76FSydBAkuf3ha2R418SGp2UPv2OhE-jU

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-1sdvr2-imfatvhn-s5&a=customerServiceUrl&ek=first_guest_reminder_01

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

Comment (zehicle) [inline rails/app/models/node.rb]: Rob Hirschfeld would like to connect on LinkedIn. How would you like to respond?

Accept: https://www.linkedin.com/e/v2?e=-1sdvr2-impb4u55-i9&a=preRegInvite&ek=second_guest_reminder_01&li=13&m=hero&ts=accept_text&sharedKey=eWcsnCis&invitationID=6118771214266753024

View Rob Hirschfeld&#39;s profile: https://www.linkedin.com/e/v2?e=-1sdvr2-impb4u55-i9&a=preRegInvite&ek=second_guest_reminder_01&li=3&m=hero&ts=profile_text&sharedKey=eWcsnCis&invitationID=6118771214266753024





You received an invitation to connect. LinkedIn will use your email address to make suggestions to our members in features like People You May Know. Unsubscribe here: https://www.linkedin.com/e/v2?e=-1sdvr2-impb4u55-i9&t=lun&midToken=AQFmrFVhGEd36Q&ek=second_guest_reminder_01&li=15&m=unsub&ts=HTML&eid=-1sdvr2-impb4u55-i9&loid=AQENI-wmQLFOgAAAAVPuBMNnmROQxGxsOdJPadx8bt8nzl6gKYNGoZLi0XSiC2-JW31m7rTLXqzI9wdxRy4jRvm-5V5EWLz639BrG_g6TU2sAGdtw2t3tn95WnIKbEDy4PK4wEzwVsBAFCca8T4Wi1wIpSjTn6zNr5kbkhKKVRA8Pczyw_Ncmoo5xMAR2PhujuYG

This email was sent to reply@reply.github.com.

If you need assistance or have questions, please contact LinkedIn Customer Service: https://www.linkedin.com/e/v2?e=-1sdvr2-impb4u55-i9&a=customerServiceUrl&ek=second_guest_reminder_01

&copy; 2016 LinkedIn Corporation, 2029 Stierlin Court, Mountain View CA 94043. LinkedIn and the LinkedIn logo are registered trademarks of LinkedIn.

**Pull request merged**